src/waypoint.jsx

```diff
@@ -333,13 +352,30 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.noWrapper) {
+      const child = React.Children.only(this.props.children);
+      const ref = (node) => {
+        this.refElement(node);
+        if (this.props.children.ref) {
+          this.props.children.ref(node);
+        }
+      };
+      return React.cloneElement(child, { ref });
+    }
+
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
+    }
+
     // We need an element that we can locate in the DOM to determine where it is
     // rendered relative to the top of its context.
     return <span ref={this.refElement} style={{ fontSize: 0 }} />;
   }
 }
 
 Waypoint.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array]),
+  noWrapper: PropTypes.bool,
```

Comment:
I'm in favor of not having the `noWrapper` prop. This is tricky though. On the one hand, I think we could go with the pragmatic choice of just adding a `div` wrapper, and solve 90+% of use-cases. But then at some point the div is going to become a problem, so maybe it's worth it to address upfront. 

I think we should remove the prop and just go wrapper-less by default, unless the children being passed in is an array, a string, or a stateless component (function). Something like:

```jsx
function getContent(children) {
  if (React.Children.count(children) > 1 || isStateless(children)) {
    return <div>{children}</div>;
  } 
  return children;
}
```

This is a simplified example, we still have to assign the `ref`. Also, I don't know how hard the `isStateless` function would be to implement.